homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -67,6 +87,39 @@ async def async_setup_entry(
     sensor = WazeTravelTime(config_entry.entry_id, name, origin, destination, data)
 
     async_add_entities([sensor], False)
+    if not hass.services.has_service(DOMAIN, SERVICE_UPDATE_TRAVEL_TIME):
+        async_setup_services(hass)
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up services for waze_travel_time integration."""
+
+    platform = entity_platform.async_get_current_platform()
+
+    async def async_update_travel_time(service_call: ServiceCall) -> ServiceResponse:
+        assert platform is not None
+        entities = await platform.async_extract_from_service(service_call)
+        if len(entities) > 0:
+            entity: WazeTravelTime = entities[0]  # type: ignore[assignment]
+
+            await hass.async_add_executor_job(entity.update)
+            return entity.extra_state_attributes
```

Comment:
I think we should document the response data e.g.   https://www.home-assistant.io/integrations/calendar/#service-calendarlist_events

Reply:
Most definitely! I will open the according documentation PR. I just wanted to get early feedback whether this approach is feasible at all.

Reply:
https://github.com/home-assistant/home-assistant.io/pull/28144